pkg/providers/aws/cluster_network_provider_test.go

```diff
@@ -2195,3 +2259,70 @@ func TestNetworkProvider_DeleteNetworkConnection(t *testing.T) {
 		})
 	}
 }
+
+func TestNetworkProvider_DeleteBundledCloudResources(t *testing.T) {
+	scheme, err := buildTestScheme()
+	if err != nil {
+		t.Fatal("failed to build scheme", err)
+	}
+	type fields struct {
+		Client         client.Client
+		RdsApi         rdsiface.RDSAPI
+		Ec2Api         ec2iface.EC2API
+		ElasticacheApi elasticacheiface.ElastiCacheAPI
+		Logger         *logrus.Entry
+	}
+	type args struct {
+		ctx context.Context
+	}
+	tests := []struct {
+		name    string
+		fields  fields
+		args    args
+		wantErr bool
+	}{
+		{
+			name: "return nil on successful deletion of bundled vpc resources",
```

Comment:
As we rely on `awsError` for `notFoundFault` should we add a test case to capture this logic. 
Also maybe a test case where we call this delete function and SubnetGroups and the SecurityGroup exist.